tiledb/sm/group/group.h

```diff
@@ -58,7 +60,10 @@ class GroupDetailsException : public StatusException {
 
 class Group {
  public:
-  Group(const URI& group_uri, StorageManager* storage_manager);
+  Group(
+      ContextResources& resources,
+      const URI& group_uri,
+      StorageManager* storage_manager);
```

Comment:
This is now a transitional constructor. It needs documentation.
- The fact that it's transitional.
- What the post-transition constructor will look like (no storage manager)
- What's required during the transition (resources come from the storage manager)